pkg/providers/v1/aws.go

```diff
@@ -619,7 +628,7 @@ func newAWSCloud2(cfg config.CloudConfig, awsServices Services, provider config.
 	}
 	awsCloud.instanceCache.cloud = awsCloud
 	awsCloud.zoneCache.cloud = awsCloud
-	awsCloud.instanceTopologyManager = newInstanceTopologyManager(awsCloud.ec2)
+	awsCloud.instanceTopologyManager = resourcemanagers.NewInstanceTopologyManager(awsCloud.ec2v2)
```

Comment:
why don't we set this when declaring the struct? does the `awsCloud` even need a reference to `ec2v2`?

Reply:
Ya, that's a good call. I killed the weird circular dependencies that we do, so no need for that.